File: android/src/main/java/com/dsr/eudi/walletkit/reactnative/model/JSEudiWalletConfig.kt
package com.dsr.eudi.walletkit.reactnative.model

import eu.europa.ec.eudi.wallet.EudiWalletConfig

data class JSEudiWalletConfig(
  val documentsStorageDir: String?,
  val useHardwareToStoreKeys: Boolean?,
  val userAuthenticationRequired: Boolean?,
  val userAuthenticationTimeOut: Double?,
  val verifyMsoPublicKey: Boolean?,
  val trustedReaderCertificates: List<String>?,
  val openId4VpConfig: JSOpenId4VpConfig?,
  val openId4VciConfig: JSOpenId4VciConfig?,
  val bleConfig: JSBLEConfig?
)

data class JSBLEConfig(
  @EudiWalletConfig.BleTransferMode
  val transferMode: Int,
  val clearCacheAfterTransfer: Boolean
)

File: ios/Sources/Model/JSEudiWalletConfig.swift
struct JSEudiWalletConfig {
    let trustedReaderCertificates: [Data]
    let userAuthenticationRequired: Bool
    let openId4VpConfig: JSOpenId4VpConfig?
    let openId4VciConfig: JSOpenId4VciConfig?
}

extension JSEudiWalletConfig {
    init(fromJson json: JSONDictionary) throws {
        let trustedReaderCertificatesPem = json["trustedReaderCertificates"] as? [String]
        let trustedReaderCertificates = try trustedReaderCertificatesPem?.map {
            try parseCertDataFromPem(certPem: $0)
        } ?? [Data]()
        
        let openId4VpConfigJson = json["openId4VpConfig"] as? JSONDictionary
        let openId4VpConfig = openId4VpConfigJson != nil ? try JSOpenId4VpConfig(fromJson: openId4VpConfigJson!) : nil
        
        
        let openId4VciConfigJson = json["openId4VciConfig"] as? JSONDictionary
        let openId4VciConfig = openId4VciConfigJson != nil ? try JSOpenId4VciConfig(fromJson: openId4VciConfigJson!) : nil
        
        self.init(
            trustedReaderCertificates: trustedReaderCertificates,
            userAuthenticationRequired: json["userAuthenticationRequired"] as? Bool ?? true,
            openId4VpConfig: openId4VpConfig,
            openId4VciConfig: openId4VciConfig
        )
    }
}
